src/server/scripts/Northrend/IcecrownCitadel/boss_professor_putricide.cpp

```diff
@@ -1658,6 +1658,27 @@ class spell_stinky_precious_decimate : public SpellScriptLoader
         }
 };
 
+// 70402, 72511, 72512, 72513 - Mutated Transformation
+class spell_abomination_mutated_transformation : public SpellScript
+{
+    PrepareSpellScript(spell_abomination_mutated_transformation);
+
+    void HandleResistance(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+        uint32 damage = GetHitDamage();
+        Unit* target = GetHitUnit();
+        damage -= Unit::CalcSpellResistedDamage(caster, target, GetHitDamage(), SPELL_SCHOOL_MASK_SHADOW, nullptr);
```

Comment:
maybe add some comments to explain what this script is doing

Reply:
https://github.com/TrinityCore/TrinityCore/issues/23837#issue-499929192

Reply:
Our current system always pick the min resist value, but this spell is a exception.
We need full shadow resist and full nature resist and apply to original damage. (check combat logs)
So... i get original damage and apply Shadow Resistence and after Nature Resistence, to get the correct new damage value.

The problem is: all damages hook happens before the damage taken on target, and when damage is taken, resistence will be applied (in this case, the min resistence will be applied again), so we need make Spell "Ignore resistences", since resistence is already calculated inside the script.

So in the end, is a script to calculate a custom resistence formula, for a exception spell.